src/api/routes/v1/get-network.ts

```diff
@@ -0,0 +1,202 @@
+      }
+      // if the value is undefined (which is returned if an error occurs)
+      if (value == null) {
+        undefinedValues += 1
+        // reject promise combinator if all promises are failed/undefined
+        if (undefinedValues + errors.length === promises.length) {
+          reject(errors)
+        }
+        return
+      }
+      resolved = true
+
+      // resolve with the first available value
+      resolve(value)
+    }
+
+    /**
+     * Helper method when a promise is rejected.
+     *
+     * @param error - The error.
+     */
+    function onError(error: Error): void {
+      // skip if already resolved
+      if (resolved) {
+        return
+      }
+
+      // collect error
+      errors = errors.concat(error)
+
+      // reject promise combinator if all promises are failed/undefined
+      if (undefinedValues + errors.length === promises.length) {
+        reject(errors)
+      }
+    }
+
+    promises.forEach(async (promise) => promise.then(onFulfill, onError))
+  })
+}
+
+/**
+ * Fetches the crawl data for the node.
+ *
+ * @param host - The host URL to crawl.
+ * @returns The crawl data from the node.
+ * @throws If none of the possible crawl ports work.
+ */
+async function fetchCrawls(host: string): Promise<Crawl> {
+  const promises: Array<Promise<Crawl | undefined>> = []
+  for (const port of CRAWL_PORTS) {
+    promises.push(crawlNode(host, port))
+  }
+  return any(promises)
+}
+
+/**
+ * Checks whether the UNL of the node is recorded in the networks table.
+ *
+ * @param unl - The UNL of the node.
+ * @returns Whether the UNL of the node has been seen before.
+ */
+async function getNetworkFromUNL(unl: string): Promise<string | undefined> {
+  const result = await query('networks')
+    .select('network')
+    .where('unls', 'like', `%${unl}%`)
+  return result.length > 0 ? result[0].network : undefined
+}
+
+/**
+ * Checks whether the public key of the node is recorded in the crawls table.
+ *
+ * @param publicKey - The public key of the node.
+ * @returns Whether the public key of the node has been seen before.
+ */
+async function getNetworkFromPublicKey(
+  publicKey: string,
+): Promise<string | undefined> {
+  const result = await query('crawls')
+    .select('public_key', 'networks')
+    .where('public_key', '=', publicKey)
+  return result.length > 0 ? result[0].networks : undefined
+}
+
+/**
+ * Adds the node (and its corresponding network) to the networks table.
+ *
+ * @param url - The URL endpoint of the node.
+ * @param unl - The UNL of the node.
+ * @returns The ID of the new network.
+ */
+async function addNode(url: string, unl: string | null): Promise<string> {
+  const currentNetworks = await query('networks')
+    .select('network')
+    .orderBy('network')
+  const maxNetwork =
+    currentNetworks[currentNetworks.length - networks.length - 1]?.network ?? 0
+  const newNetwork = (Number(maxNetwork) + 1).toString()
+  const network: Network = {
+    network: newNetwork,
```

Comment:
Nit: I got a bit confused reading this because I couldn't tell if `Network.network` was supposed to be an integer or a string. Maybe a better name for that field would be `id`?

Reply:
It's referred to as `network` in a bunch of the existing tables, which is why I kept that as `id`. But happy to rename it in this one if you'd prefer.

Reply:
It's actually a string (it's `main`/`test/`dev/`nft-dev` for the existing networks, and I didn't want to mess with the existing tables too much bc migrations by hand are hard and scary). But I thought the best way to add a new network without any details about it would be to just assign it a number.

Reply:
`currentNetworks.length - networks.length - 1` handles that - non-number strings will be after the number strings when sorted alphabetically

Reply:
Hm, looks like I'll have to rethink the sorting anyways - I forgot that with strings, '2' comes after '11'

Reply:
I think it'd be worth adding a comment explaining that going to `currentNetworks.length - networks.length - 1` is to go to the end of the named networks (`networks.length`) and into the numbered networks.